pkg/cvo/status.go

```diff
@@ -180,8 +181,16 @@ func (optr *Operator) syncStatus(ctx context.Context, original, config *configv1
 
 	now := metav1.Now()
 	version := versionString(status.Actual)
+	desired := configv1.Release{
+		Version: status.Actual.Version,
+		Image:   status.Actual.Image,
+	}
+	if availableUpdates != nil && desired.Image == availableUpdates.Current.Image {
```

Comment:
Should check `availableUpdates` above where first set.

Reply:
There's no such guard [in master today][1], because `NeedsUpdate` [has a `nil` guard][2].  So I think we're ok.  This call is just "hey, let's look inside that `availableUpdates` we were getting before and, if the images match, flesh out `desired` with any available data.  Although... I guess that will miss information from the release image itself.  We probably want to use `optr.currentVersion()` if that's an image-match instead.

[1]: https://github.com/openshift/cluster-version-operator/blob/e49df43aaa578f0f420337bc28b7fc0f8cf35c09/pkg/cvo/status.go#L163
[2]: https://github.com/openshift/cluster-version-operator/blob/e49df43aaa578f0f420337bc28b7fc0f8cf35c09/pkg/cvo/availableupdates.go#L91-L93

Reply:
Yeah, we went through this availableUpdates being nil thing before. I checked it and as I recall I determined that it can only be nil at startup/initialization which is probably why it wasn't originally being checked in here. I just saw that you had checked it down below which stood out so I'd pull that check.

Reply:
>  Although... I guess that will miss information from the release image itself...

I've pushed 33333ecb -> c1de00c3, pulling `mergeReleaseMetadata` out into a separate helper (with tests), pushing `Release`-from-payload calculations down into `pkg/payload` itself (so both of our `LoadUpdate` consumers can share it), and adding a `mergeReleaseMetadata` call in `syncStatus` to catch cases like "I'm updating to a new release, and still want to layer in any metadata from `availableUpdates`".